central/processlisteningonport/store/postgres/full_store.go

```diff
@@ -31,7 +31,7 @@ type fullStoreImpl struct {
 // process_indicators. Used to provide information for queries like 'give
 // me all PLOP by this deployment'.
 // XXX: Verify the query plan to make sure needed indexes are in use.
-const getByDeploymentStmt = "SELECT plop.id, plop.serialized, " +
+const getByDeploymentStmt = "SELECT plop.id, plop.poduid, plop.serialized, " +
```

Comment:
plop.id doesn't seem to be used anywhere. Do we need it?

Reply:
Really, based on usage, the only thing this query needs is `plop.serialized`.  The way the stores typically work is to get serialized and unmarshall it.  We should stick to that pattern here.  We shouldn't grab the `poduid` from the table, but from the serialized object.  Based on our typical access patterns we should only use the `poduid` in the where clause OR if we are using the query framework to target specific columns.  In this case we are getting all the data, so we should process all that data the same way.  Does that make sense?

Reply:
You are suggesting that poduid, be stored in the serialized column, instead of being its own column? I have a PR that I am working on to delete listening endpoints when the pod is deleted. I think that requires that poduid be its own column, instead of being stored in the serialized data.

My question was about plop.id not plop.poduid. It looks like plop.id is not used anywhere, so I don't think it should be returned here. I was thinking of removing it, but decided to leave the question here first.

Reply:
The other PR that I referred to above is this one https://github.com/stackrox/stackrox/pull/8020

It is not yet ready to be reviewed, but I think it is getting close.

Reply:
To clarify, everything is stored in the serialized object.  Whether it is a column on its own or not, it is in the serialized object.  If you look in the store logic, you will see that any query of the objects only use the serialized object when reading the rows.  You can do the same thing here.  The only exception is when using the query framework to pull specific items which is not the case here as we are dealing with the entire object.

I agree that the `poduid` should be a column.  What I'm saying is that it does not need to be in this query.  Really `plop.id` falls under the same rule.  The query could simply be `select serialized` and let the unmarshalling populate all the fields.

Reply:
The query is now only returning the serialized columns.